packages/cashier-v2/src/components/NewsTicker/NewsTicker.tsx

```diff
@@ -0,0 +1,47 @@
+import React, { useState } from 'react';
+import clsx from 'clsx';
+import NewsTickerChildren from './NewsTickerChildren';
+import styles from './NewsTicker.module.scss';
+
+type TNewsTicker = {
+    children: React.ReactNode;
+    className?: string;
+    speed: number;
+};
+
+const NewsTicker: React.FC<TNewsTicker> = ({ children, className, speed }) => {
+    const [elementWidth, setElementWidth] = useState(-1);
+    const [isExceedingParent, setIsExceedingParent] = useState(false);
+
+    const onRefChange = (ref: HTMLDivElement) => {
+        if (ref) {
+            setIsExceedingParent(ref.scrollWidth > ref.clientWidth);
+
+            if (elementWidth === -1) {
+                setElementWidth(ref.scrollWidth);
+            }
+        }
+    };
+
+    const animationDuration = elementWidth / speed; // time = distance / speed
```

Comment:
out of curiosity - is it necessary to be dynamic? could it be static and just in .css, potentially under @media query to differentiate between mobile and desktop if needed (basically - the less css in javascript the better :D)

Reply:
sure @wojciech-deriv 
we can fix it in upcoming PR's
thanks for pointing on this